packages/service-worker/src/low_level.ts

```diff
@@ -165,27 +175,26 @@ export class NgswCommChannel {
 
       const rawEvents = fromEvent<MessageEvent>(serviceWorker, 'message');
       const rawEventPayload = rawEvents.pipe(map((event) => event.data));
-      const eventsUnconnected = rawEventPayload.pipe(filter((event) => event && event.type));
-      const events = eventsUnconnected.pipe(publish()) as ConnectableObservable<IncomingEvent>;
-      events.connect();
+      const events = rawEventPayload.pipe(filter((event) => event && event.type));
```

Comment:
```suggestion
      const _events = new Subject<TypedEvent>();
      this.events = _events.asObservable();
      const messageListener = (event: MessageEvent) => {
        const {data} = event;
        if (data?.type) {
          _events.next(data);
        }
      };
      serviceWorker.addEventListener('message', messageListener);

      // The injector is optional to avoid breaking changes.
      const destroyRef = injector?.get(DestroyRef, null, {optional: true});
      destroyRef?.onDestroy(() => {
        serviceWorker.removeEventListener('controllerchange', updateController);
        serviceWorker.removeEventListener('message', messageListener);
      });
```

I think we can go a fair bit further without too much difficulty. WDYT?